stellar-contract-env-host/src/test.rs

```diff
@@ -352,3 +361,101 @@ fn vec_append_empty() {
+    ];
+    let id = ContractID([0; 32].into());
+    let vm = Vm::new(&host, id, &code).unwrap();
+    let a = 4i32;
+    let b = 7i32;
+    let c = 0x7fffffff_i32;
+    let scvec0: ScVec = ScVec(vec![ScVal::I32(a), ScVal::I32(b)].try_into()?);
+    let res = vm.invoke_function(&host, "add", &scvec0).unwrap();
+    match res {
+        ScVal::I32(v) => assert_eq!(v, a + b),
+        _ => panic!("Wrong result type"),
+    }
+    // overflow
+    let scvec0: ScVec = ScVec(vec![ScVal::I32(a), ScVal::I32(c)].try_into()?);
+    let res = catch_unwind(AssertUnwindSafe(|| {
+        vm.invoke_function(&host, "add", &scvec0).unwrap();
+    }));
+    assert!(res.is_err());
+    Ok(())
+}
+
+#[cfg(feature = "vm")]
+#[test]
+fn contract_invoke_another_contract() -> Result<(), ()> {
+    use im_rc::OrdMap;
+
+    let contract_id = ContractID([0; 32].into());
+    let key = ScVal::Static(ScStatic::Void); // TODO: replace with "SCS_LEDGER_KEY_CONTRACT_CODE_WASM"
+    let storage_key = Key { contract_id, key };
+    let code: [u8; 163] = [
+        0x00, 0x61, 0x73, 0x6d, 0x01, 0x00, 0x00, 0x00, 0x01, 0x07, 0x01, 0x60, 0x02, 0x7e, 0x7e,
+        0x01, 0x7e, 0x03, 0x02, 0x01, 0x00, 0x05, 0x03, 0x01, 0x00, 0x10, 0x06, 0x11, 0x02, 0x7f,
+        0x00, 0x41, 0x80, 0x80, 0xc0, 0x00, 0x0b, 0x7f, 0x00, 0x41, 0x80, 0x80, 0xc0, 0x00, 0x0b,
+        0x07, 0x2b, 0x04, 0x06, 0x6d, 0x65, 0x6d, 0x6f, 0x72, 0x79, 0x02, 0x00, 0x03, 0x61, 0x64,
+        0x64, 0x00, 0x00, 0x0a, 0x5f, 0x5f, 0x64, 0x61, 0x74, 0x61, 0x5f, 0x65, 0x6e, 0x64, 0x03,
+        0x00, 0x0b, 0x5f, 0x5f, 0x68, 0x65, 0x61, 0x70, 0x5f, 0x62, 0x61, 0x73, 0x65, 0x03, 0x01,
+        0x0a, 0x47, 0x01, 0x45, 0x01, 0x02, 0x7f, 0x02, 0x40, 0x20, 0x00, 0x42, 0x0f, 0x83, 0x42,
+        0x03, 0x52, 0x20, 0x01, 0x42, 0x0f, 0x83, 0x42, 0x03, 0x52, 0x72, 0x45, 0x04, 0x40, 0x20,
+        0x01, 0x42, 0x04, 0x88, 0xa7, 0x22, 0x02, 0x41, 0x00, 0x48, 0x20, 0x02, 0x20, 0x00, 0x42,
+        0x04, 0x88, 0xa7, 0x22, 0x03, 0x6a, 0x22, 0x02, 0x20, 0x03, 0x48, 0x73, 0x45, 0x0d, 0x01,
+        0x0b, 0x00, 0x0b, 0x20, 0x02, 0xad, 0x42, 0x04, 0x86, 0x42, 0x03, 0x84, 0x0b,
+    ];
+    let scob = ScObject::Binary(code.try_into()?);
+    let val = ScVal::Object(Some(Box::new(scob)));
+    let map = OrdMap::unit(storage_key.clone(), Some(val));
+    let mut footprint = Footprint::default();
+    footprint.record_access(&storage_key, AccessType::ReadOnly);
+
+    let storage = Storage::with_enforcing_footprint_and_map(footprint, map);
+    let host = Host::with_storage(storage);
+
+    let sym = Symbol::from_str("add");
+    let scobj = ScObject::Binary([0; 32].try_into()?);
+    let obj = host.to_host_obj(&scobj).unwrap();
+    let res = host
+        .call2(obj.to_object(), sym.into(), 1i32.into(), 2i32.into())
+        .unwrap();
+
+    assert!(res.is::<i32>());
+    assert!(res.get_tag() == Tag::I32);
+    let i = unsafe { <i32 as RawValConvertible>::unchecked_from_val(res) };
```

Comment:
I think you can do `let i : i32 = res.try_into().unwrap()`

Reply:
Thanks, fixed.